### PR TITLE
Bump test package dependencies

### PR DESCRIPTION
**Pull request by jiahaog**
## Description

Update to the following:

- test: 1.16.0-nullsafety.6
- test_api: 0.2.19-nullsafety.3
- test_core: 0.3.12-nullsafety.6

Motivation: We want to use a the new APIs provided in the new version of `package:test_core` in `package:integration_test`, but there is a dependency conflict. This is because `package:integration_test` depends on `package:flutter_test` and `package:flutter_driver`, both of which also transitively depends on a conflicting pinned version of `package:test_core.

I'm not too familiar with the process for doing this, let me know if this isn't correct. These are the steps I took:

1. Update the pinned versions in `packages/flutter_tools/lib/src/commands/update_packages.dart`
2. Update the pinned versions in `packages/flutter_tools/pubspec.yaml` 
3. Move `http` in `dev/devicelab/pubspec.yaml` from `dev_dependencies` to `dependencies`. This is needed because it's actually used as a normal dependency. The new `package:test` `1.16.0-nullsafety.6` [removes](https://github.com/dart-lang/test/commit/0ac497bb94c5708c2fd24c83f9f6acaf235296e3) the dependency on `package:http`. If I run the `update-packages` command without doing so, `http` will be removed from `dev_dependencies`, breaking the build.
4. Run `flutter update-packages --force-upgrade`.

### CODE REVIEW

File: dev/devicelab/pubspec.yaml [hunk 1]
```diff
@@ -10,6 +10,7 @@ environment:
 dependencies:
   args: 1.6.0
   file: 6.0.0-nullsafety.2
+  http: 0.12.2
```
Comment (natebosch) [inline dev/devicelab/pubspec.yaml:13]: Had this been missing before?

**Review by natebosch — APPROVED** [review]

File: dev/devicelab/pubspec.yaml (hunk 1 above)
Comment (jiahaog) [reply dev/devicelab/pubspec.yaml:13]: Yeah, I had to move this manually from `dev_dependencies` below to here before running `flutter update-packages --force-upgrade`. I believe the removal of package:http from package:test 1.16.0-nullsafety.6 causes the command to remove it from `dev_dependencies` below, and it actually should be a dependency.

**Comment by jonahwilliams** [comment]
Unfortunately it looks like someone else updated packages before you? You'll need to rebase first.

**Comment by jiahaog** [comment]
No worries, the conflicting PR that was merged already updates the dependencies needed here.

**Pull request closed without merging**